web/src/filter.ts

```diff
@@ -1619,6 +1621,9 @@ export class Filter {
 
         const adjusted_terms = Filter.adjusted_terms_if_moved(this._terms, message);
         if (adjusted_terms) {
+            // If the narrow terms are adjusted, then we need to update the
+            // hash user entered, to point to the updated narrow.
+            this.narrow_requires_hash_change = true;
```

Comment:
Does something clear this back to false after the hash is updated?

Reply:
For one particular filter, it is set to false via the constructor,  and set to true if narrow adjusted for first time. This is necessary irrespective of whether message found locally because we need to update the hash to match the narrow

 For other narrows, other filter instances are created.

Reply:
https://github.com/zulip/zulip/pull/30887/files#r1680545184